sql/mssql/inspect.go

```diff
@@ -110,6 +107,62 @@ type queryScope struct {
 	append func(*schema.Schema, string, *schema.Column) error
 }
 
+// checks queries and appends the check constraints of the given table.
+func (i *inspect) checks(ctx context.Context, s *schema.Schema) error {
+	rows, err := i.querySchema(ctx, checksQuery, s)
+	if err != nil {
+		return fmt.Errorf("mssql: querying schema %q check constraints: %w", s.Name, err)
+	}
+	defer rows.Close()
+	if err := i.addChecks(s, rows); err != nil {
+		return err
+	}
+	return rows.Err()
+}
+
+// addChecks scans the rows and adds the checks to the table.
+func (i *inspect) addChecks(s *schema.Schema, rows *sql.Rows) error {
+	names := make(map[string]*schema.Check)
+	for rows.Next() {
+		var (
+			table, name, clause string
+			column              sql.NullString
+			disabled            int64
+		)
+		if err := rows.Scan(&table, &name, &clause, &column, &disabled); err != nil {
+			return fmt.Errorf("mssql: scanning check: %w", err)
+		}
+		t, ok := s.Table(table)
+		if !ok {
+			return fmt.Errorf("mssql: table %q was not found in schema", table)
+		}
+		if column.Valid {
+			if _, ok := t.Column(column.String); !ok {
+				return fmt.Errorf("mssql: column %q was not found for check %q", column.String, name)
+			}
+		}
+		check, ok := names[name]
+		if !ok {
+			check = &schema.Check{
+				Name:  name,
+				Expr:  clause,
+				Attrs: nil,
+			}
+			if column.Valid {
+				check.Attrs = append(check.Attrs, &CheckColumns{
+					Columns: []string{column.String},
+				})
+			}
+			if disabled == 1 {
```

Comment:
Can’t you use sql.NullBool instead, or just bool?

Reply:
I will update in next PR for other places too.